python/cudf/cudf/core/dataframe.py

```diff
@@ -1205,14 +1214,21 @@ def _concat(
 
         # Concatenate the Tables
         out = cls._from_data(
-            *libcudf.concat.concat_tables(tables, ignore_index)
+            *libcudf.concat.concat_tables(
+                tables, ignore_index=ignore_index or are_all_range_index
+            )
         )
 
         # If ignore_index is True, all input frames are empty, and at
         # least one input frame has an index, assign a new RangeIndex
         # to the result frame.
         if empty_has_index and num_empty_input_frames == len(objs):
             out._index = cudf.RangeIndex(result_index_length)
+        elif are_all_range_index and not ignore_index:
+            out._index = cudf.core.index.GenericIndex._concat(
+                [o._index for o in objs]
+            )
```

Comment:
Is this case not included in line 1218?

Reply:
Nope, that line's expectation is to have the index columns materialized. Whereas we don't want to materialize and hit the specialized concat rangeindex logic in index.py

Reply:
Especially we want to hit `_concat_range_index` in this case: 
https://github.com/rapidsai/cudf/blob/eab2486008229414737de8107d04fef63d5a1dd7/python/cudf/cudf/core/index.py#L2311